app/services/room_settings_getter.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class RoomSettingsGetter
+  # Special options are meeting options that have different values other than `true|false` to represent postives and negatives.
+  # The `special_options` hash is a registry that keeps hold of all of the special options used in GL3 along with their postive and negative values.
+  # When adding a new special option just register it in this Hash respecting the fallowing format:
+  # Hash(`<option_name> => {'true' => <Postive>, 'false' => <Negative>})`
+  SPECIAL_OPTIONS = { 'guestPolicy' => { 'true' => 'ASK_MODERATOR', 'false' => 'ALWAYS_ACCEPT' } }.freeze
+
+  def initialize(room_id:, provider:, only_bbb_options: false)
+    @room_id = room_id
+    @only_bbb_options = only_bbb_options
+    # Fetching only rooms configs that are not optional to overwrite the settings values.
+    @rooms_configs = MeetingOption.joins(:rooms_configurations)
+                                  .where(rooms_configurations: { provider: })
+                                  .where.not(rooms_configurations: { value: 'optional' })
+                                  .pluck(:name, :value)
+                                  .to_h
+  end
+
+  def call
+    room_settings = MeetingOption.joins(:room_meeting_options).where(room_meeting_options: { room_id: @room_id })
+    room_settings = room_settings.where.not('name ILIKE :prefix', prefix: 'gl%') if @only_bbb_options
+    room_settings = room_settings.pluck(:name, :value).to_h
+
+    room_settings.merge!(@rooms_configs) # Merging rooms settings with its none optional configurations prioritizing forced configs over settings.
+    room_settings_special_options = SPECIAL_OPTIONS.slice(*room_settings.keys) # Extracting the room settings special options to minimize iterations.
+
+    # Configs are enum of ['true', 'false', 'optional'], after merging none optional settings with their configs only the room
+    # special options have to map positive/negative values.
+    room_settings_special_options.each_key do |name|
+      config = room_settings[name]
+      room_settings[name] = SPECIAL_OPTIONS[name][config] if %w[true false].include? config # Config values are expected to be 'true'|'false'
+    end
```

Comment:
This is still super confusing to me - I'm having a hard time understanding the complexity behind this. What's the goal of this? To replace `ASK_MODERATOR` with `true`?

Reply:
That's the result of our little optimization.

The service will:
1. Fetch **only** the rooms configs  that are marked  as **not** optional (`true|false`) as a `Hash(<setting_name>=><'true'|'false'>)`.
2. Fetch **all** of the the room settings as a `Hash(<setting_name> => <value>)`.
3. Merge the room settings hash with the rooms configs.
This step will overwrite all none optional configured settings with their forced value `true|false` (including but not limited to `ASK_MODERATOR`).
4. Filter only the room special settings as a `Hash(<setting_name> => Hash('true'=><POSITIVE>, 'false' => <NEGATIVE>))`.
5. Since only the room special settings aren't ready by nature and require mapping of the forced values into their expected output, the room special settings hash will be mapped where each setting (including but not limited to `ASK_MODERATOR`) existing on the room settings and the  `SPECIAL_OPTIONS` will map the setting to its **positive/negative** value.

So, when we merge the rooms settings with the rooms configs we will overwrite their values.
That optimizes our work because only a handful of  settings that we call "special" are only those which requires an extra step to set properly.
All optional settings (even the special ones) will conserve their values, while regular none optional settings will have their forced binary value and none optional special settings will map will map their values to its final output as registered in the `SPECIAL_OPTIONS` constant and according to its small specification.

Hope that clarifies the changes,